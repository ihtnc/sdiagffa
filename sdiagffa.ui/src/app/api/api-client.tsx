const baseUrl = 'https://localhost:44300';

export interface Person {
  id: number,
  name: string,
  gender: string,
  birthYear: string,
  height: string,
  mass: string,
  skinColor: string,
  eyeColor: string
};

export interface Film {
  id: number,
  title: string,
  releaseDate: string,
  episodeId: number,
  director: string,
  producer: string
};

export interface Planet {
  id: number,
  name: string,
  terrain: string,
  climate: string,
  surfaceWater: string,
  gravity: string,
  population: string,
  diameter: string
};

export interface Connection {
  personId: number,
  name: string,
  connection: 'Film' | 'Homeworld' | 'None',
  connectionId: number | null,
  connectionName: string | null,
  next: Connection | null
};

export interface PagedResults<T> {
  count: number,
  nextPageUrl: string | null,
  previousPageUrl: string | null,
  results: Array<T>
};

export const searchPeople = async function (search: string, page: number | null = null)
  : Promise<PagedResults<Person>> {
  const findUrl = '/details/people';

  const queries = [];
  if (search.length > 0) { queries.push(`search=${search}`); }
  if (page !== null) { queries.push(`page=${page}`); }
  const queryString = queries.length > 0 ? `?${queries.join('&')}` : '';

  const data = await fetch(`${baseUrl}${findUrl}${queryString}`);
  return data.json() as unknown as PagedResults<Person>;
};

export const customPeopleSearch = async function (peopleSearchUrl: string)
  : Promise<PagedResults<Person>> {
  const data = await fetch(`${baseUrl}${peopleSearchUrl}`);
  return data.json() as unknown as PagedResults<Person>;
};

export const findConnection = async function (referencePersonId: number, targetPersonId: number)
  : Promise<Connection> {
    const findConnectionUrl = '/find-connection';
    const requestOptions = {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({
        referencePersonId,
        targetPersonId
      })
    };
    const data = await fetch(`${baseUrl}${findConnectionUrl}`, requestOptions);
    return data.json() as unknown as Connection;
};

export const getPerson = async function (id: number)
  : Promise<Person> {
  const getUrl = '/details/people';

  const data = await fetch(`${baseUrl}${getUrl}/${id}`);
  return data.json() as unknown as Person;
};


export const getFilm = async function (id: number)
  : Promise<Film> {
  const getUrl = '/details/films';

  const data = await fetch(`${baseUrl}${getUrl}/${id}`);
  return data.json() as unknown as Film;
};

export const getPlanet = async function (id: number)
  : Promise<Planet> {
  const getUrl = '/details/planets';

  const data = await fetch(`${baseUrl}${getUrl}/${id}`);
  return data.json() as unknown as Planet;
};