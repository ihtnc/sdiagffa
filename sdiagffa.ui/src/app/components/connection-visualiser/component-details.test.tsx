import { beforeEach, afterEach, describe, expect, it, vi } from "vitest";
import { render, screen, waitFor, waitForElementToBeRemoved } from '@testing-library/react'
import * as ApiClient from "@/api/api-client";
import ConnectionDetails from "./connection-details";
import useGetDetails, { Details, DetailsType } from "./get-details-store";

vi.mock('usehooks-ts', async () => {
  const useOnClickOutside = vi.fn();
  const useBoolean = (await vi.importActual<typeof import('usehooks-ts')>('usehooks-ts')).useBoolean;
  return { useOnClickOutside, useBoolean };
});

describe('ConnectionDetails component', () => {
  let getPersonSpy;
  let getFilmSpy;
  let getPlanetSpy;

  beforeEach(() => {
    getPersonSpy = vi.spyOn(ApiClient, 'getPerson');
    getFilmSpy = vi.spyOn(ApiClient, 'getFilm');
    getPlanetSpy = vi.spyOn(ApiClient, 'getPlanet');
    useGetDetails.setState({ detailsCache: [] })
  });

  afterEach(() => {
    getPersonSpy.mockReset();
    getFilmSpy.mockReset();
    getPlanetSpy.mockReset();
  });

  it('should render default state', () => {
    getPersonSpy.mockReturnValue({ });
    const component = render(<ConnectionDetails id={1} type={DetailsType.Character} />);
    expect(component).toMatchSnapshot();
  });

  it('should call getPerson on initial load if type is Character', () => {
    getPersonSpy.mockReturnValue({});
    expect(getPersonSpy).not.toHaveBeenCalled();

    const id = 1;
    const type = DetailsType.Character;
    render(<ConnectionDetails id={id} type={type} />);

    expect(getPersonSpy).toHaveBeenCalledWith(id);
  });

  it('should call getFilm on initial load if type is Film', () => {
    getFilmSpy.mockReturnValue({});
    expect(getFilmSpy).not.toHaveBeenCalled();

    const id = 1;
    const type = DetailsType.Film;
    render(<ConnectionDetails id={id} type={type} />);

    expect(getFilmSpy).toHaveBeenCalledWith(id);
  });

  it('should call getPlanet on initial load if type is Homeworld', () => {
    getPlanetSpy.mockReturnValue({});
    expect(getPlanetSpy).not.toHaveBeenCalled();

    const id = 1;
    const type = DetailsType.Homeworld;
    render(<ConnectionDetails id={id} type={type} />);

    expect(getPlanetSpy).toHaveBeenCalledWith(id);
  });

  it('should render character details', async () => {
    var person: ApiClient.Person = {
      id: 1,
      name: 'Luke Skywalker',
      birthYear: '19BBY',
      eyeColor: 'blue',
      gender: 'male',
      height: '172',
      mass: '77',
      skinColor: 'fair'
    };
    getPersonSpy.mockResolvedValue(person);

    const id = 1;
    const type = DetailsType.Character;
    const component = render(<ConnectionDetails id={id} type={type} />);

    await waitFor(() => {
      expect(screen.getByText('Luke Skywalker')).toBeInTheDocument();
    });

    expect(component).toMatchSnapshot();
    const icon = component.container.querySelector('.fa-user');
    expect(icon).toBeInTheDocument();
  });

  it('should render film details', async () => {
    var film: ApiClient.Film = {
      id: 1,
      title: 'A New Hope',
      episodeId: 4,
      director: 'George Lucas',
      producer: 'Gary Kurtz, Rick McCallum',
      releaseDate: '1977-05-25'
    };
    getFilmSpy.mockResolvedValue(film);

    const id = 1;
    const type = DetailsType.Film;
    const component = render(<ConnectionDetails id={id} type={type} />);

    await waitFor(() => {
      expect(screen.getByText('A New Hope')).toBeInTheDocument();
    });

    expect(component).toMatchSnapshot();
    const icon = component.container.querySelector('.fa-film');
    expect(icon).toBeInTheDocument();
  });

  it('should render planet details', async () => {
    var planet: ApiClient.Planet = {
      id: 1,
      name: 'Tatooine',
      climate: 'arid',
      terrain: 'desert',
      surfaceWater: '1',
      diameter: '10465',
      gravity: '1 standard',
      population: '200000'
    };
    getPlanetSpy.mockResolvedValue(planet);

    const id = 1;
    const type = DetailsType.Homeworld;
    const component = render(<ConnectionDetails id={id} type={type} />);

    await waitFor(() => {
      expect(screen.getByText('Tatooine')).toBeInTheDocument();
    });

    expect(component).toMatchSnapshot();
    const icon = component.container.querySelector('.fa-globe');
    expect(icon).toBeInTheDocument();
  });

  it('should retrieve details when id property changes', () => {
    getPersonSpy.mockReturnValueOnce({ id: 1 });
    getPersonSpy.mockReturnValueOnce({ id: 2 });

    const id = 1;
    const type = DetailsType.Character;
    const { rerender } = render(<ConnectionDetails id={id} type={type} />);
    expect(getPersonSpy).toHaveBeenCalledWith(id);

    const newId = 2;
    rerender(<ConnectionDetails id={newId} type={type} />)
    expect(getPersonSpy).toHaveBeenCalledWith(newId);
  });

  it('should retrieve details when type property changes', () => {
    getPersonSpy.mockReturnValueOnce({ id: 1 });
    getFilmSpy.mockReturnValueOnce({ id: 1 });

    const id = 1;
    const type = DetailsType.Character;
    const { rerender } = render(<ConnectionDetails id={id} type={type} />);
    expect(getPersonSpy).toHaveBeenCalledWith(id);

    const newType = DetailsType.Film;
    rerender(<ConnectionDetails id={id} type={newType} />)
    expect(getFilmSpy).toHaveBeenCalledWith(id);
  });

  it('should not retrieve details when id was previously retrieved', () => {
    const details: Array<Details> = [{
      id: 1,
      type: DetailsType.Character,
      heading: '',
      additionalDetails: { }
    }];
    useGetDetails.setState({ detailsCache: details});

    getPersonSpy.mockReturnValueOnce({ id: 1 });

    const id = 1;
    const type = DetailsType.Character;
    render(<ConnectionDetails id={id} type={type} />);

    expect(getPersonSpy).not.toHaveBeenCalled();
  });

  it('should render when clicked outside', async () => {
    const useHooks = await import('usehooks-ts');
    let clickOutsideCallback;
    useHooks.useOnClickOutside = vi.fn().mockImplementation((_, func) => {
      clickOutsideCallback = func;
    });

    var person: ApiClient.Person = {
      id: 1,
      name: 'Luke Skywalker',
      birthYear: '19BBY',
      eyeColor: 'blue',
      gender: 'male',
      height: '172',
      mass: '77',
      skinColor: 'fair'
    };
    getPersonSpy.mockResolvedValue(person);

    const component = render(<ConnectionDetails id={1} type={DetailsType.Character} />);

    await waitFor(() => {
      expect(screen.getByText('Luke Skywalker')).toBeInTheDocument();
    });

    clickOutsideCallback();

    await waitForElementToBeRemoved(() => screen.getByText('Luke Skywalker'));

    expect(component).toMatchSnapshot();
  });
});