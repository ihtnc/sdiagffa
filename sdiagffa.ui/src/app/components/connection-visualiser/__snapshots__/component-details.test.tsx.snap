// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ConnectionDetails component > should render character details 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="
        absolute top-2 left-2
        text-sm
        rounded-md
        border-2
        border-swblue dark:border-swyellow
        max-w-60 min-w-60
        bg-swgray/90 dark:bg-swblack/90
      "
      >
        <div
          class="p-2
        font-semibold
        text-white dark:text-swblack
        bg-swblue dark:bg-swyellow
        flex justify-between items-stretch
      "
        >
          <span>
            Luke Skywalker
          </span>
          <i
            class="fa fa-user self-center"
          />
        </div>
        <div
          class="p-2"
        >
          <div
            class="flex"
          >
            <label
              class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
            >
              type
            </label>
            <span
              class="w-1/2
              capitalize
              text-black dark:text-white"
            >
              Character
            </span>
          </div>
          <div
            class="flex"
          >
            <label
              class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
            >
              gender
            </label>
            <span
              class="w-1/2
              capitalize
              text-black dark:text-white"
            >
              male
            </span>
          </div>
          <div
            class="flex"
          >
            <label
              class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
            >
              birthYear
            </label>
            <span
              class="w-1/2
              capitalize
              text-black dark:text-white"
            >
              19BBY
            </span>
          </div>
          <div
            class="flex"
          >
            <label
              class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
            >
              height
            </label>
            <span
              class="w-1/2
              capitalize
              text-black dark:text-white"
            >
              172
            </span>
          </div>
          <div
            class="flex"
          >
            <label
              class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
            >
              mass
            </label>
            <span
              class="w-1/2
              capitalize
              text-black dark:text-white"
            >
              77
            </span>
          </div>
          <div
            class="flex"
          >
            <label
              class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
            >
              skinColor
            </label>
            <span
              class="w-1/2
              capitalize
              text-black dark:text-white"
            >
              fair
            </span>
          </div>
          <div
            class="flex"
          >
            <label
              class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
            >
              eyeColor
            </label>
            <span
              class="w-1/2
              capitalize
              text-black dark:text-white"
            >
              blue
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="
        absolute top-2 left-2
        text-sm
        rounded-md
        border-2
        border-swblue dark:border-swyellow
        max-w-60 min-w-60
        bg-swgray/90 dark:bg-swblack/90
      "
    >
      <div
        class="p-2
        font-semibold
        text-white dark:text-swblack
        bg-swblue dark:bg-swyellow
        flex justify-between items-stretch
      "
      >
        <span>
          Luke Skywalker
        </span>
        <i
          class="fa fa-user self-center"
        />
      </div>
      <div
        class="p-2"
      >
        <div
          class="flex"
        >
          <label
            class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
          >
            type
          </label>
          <span
            class="w-1/2
              capitalize
              text-black dark:text-white"
          >
            Character
          </span>
        </div>
        <div
          class="flex"
        >
          <label
            class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
          >
            gender
          </label>
          <span
            class="w-1/2
              capitalize
              text-black dark:text-white"
          >
            male
          </span>
        </div>
        <div
          class="flex"
        >
          <label
            class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
          >
            birthYear
          </label>
          <span
            class="w-1/2
              capitalize
              text-black dark:text-white"
          >
            19BBY
          </span>
        </div>
        <div
          class="flex"
        >
          <label
            class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
          >
            height
          </label>
          <span
            class="w-1/2
              capitalize
              text-black dark:text-white"
          >
            172
          </span>
        </div>
        <div
          class="flex"
        >
          <label
            class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
          >
            mass
          </label>
          <span
            class="w-1/2
              capitalize
              text-black dark:text-white"
          >
            77
          </span>
        </div>
        <div
          class="flex"
        >
          <label
            class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
          >
            skinColor
          </label>
          <span
            class="w-1/2
              capitalize
              text-black dark:text-white"
          >
            fair
          </span>
        </div>
        <div
          class="flex"
        >
          <label
            class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
          >
            eyeColor
          </label>
          <span
            class="w-1/2
              capitalize
              text-black dark:text-white"
          >
            blue
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ConnectionDetails component > should render default state 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ConnectionDetails component > should render film details 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="
        absolute top-2 left-2
        text-sm
        rounded-md
        border-2
        border-swblue dark:border-swyellow
        max-w-60 min-w-60
        bg-swgray/90 dark:bg-swblack/90
      "
      >
        <div
          class="p-2
        font-semibold
        text-white dark:text-swblack
        bg-swblue dark:bg-swyellow
        flex justify-between items-stretch
      "
        >
          <span>
            A New Hope
          </span>
          <i
            class="fa fa-film self-center"
          />
        </div>
        <div
          class="p-2"
        >
          <div
            class="flex"
          >
            <label
              class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
            >
              type
            </label>
            <span
              class="w-1/2
              capitalize
              text-black dark:text-white"
            >
              Film
            </span>
          </div>
          <div
            class="flex"
          >
            <label
              class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
            >
              releaseDate
            </label>
            <span
              class="w-1/2
              capitalize
              text-black dark:text-white"
            >
              1977-05-25
            </span>
          </div>
          <div
            class="flex"
          >
            <label
              class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
            >
              episodeId
            </label>
            <span
              class="w-1/2
              capitalize
              text-black dark:text-white"
            >
              4
            </span>
          </div>
          <div
            class="flex"
          >
            <label
              class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
            >
              director
            </label>
            <span
              class="w-1/2
              capitalize
              text-black dark:text-white"
            >
              George Lucas
            </span>
          </div>
          <div
            class="flex"
          >
            <label
              class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
            >
              producer
            </label>
            <span
              class="w-1/2
              capitalize
              text-black dark:text-white"
            >
              Gary Kurtz, Rick McCallum
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="
        absolute top-2 left-2
        text-sm
        rounded-md
        border-2
        border-swblue dark:border-swyellow
        max-w-60 min-w-60
        bg-swgray/90 dark:bg-swblack/90
      "
    >
      <div
        class="p-2
        font-semibold
        text-white dark:text-swblack
        bg-swblue dark:bg-swyellow
        flex justify-between items-stretch
      "
      >
        <span>
          A New Hope
        </span>
        <i
          class="fa fa-film self-center"
        />
      </div>
      <div
        class="p-2"
      >
        <div
          class="flex"
        >
          <label
            class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
          >
            type
          </label>
          <span
            class="w-1/2
              capitalize
              text-black dark:text-white"
          >
            Film
          </span>
        </div>
        <div
          class="flex"
        >
          <label
            class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
          >
            releaseDate
          </label>
          <span
            class="w-1/2
              capitalize
              text-black dark:text-white"
          >
            1977-05-25
          </span>
        </div>
        <div
          class="flex"
        >
          <label
            class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
          >
            episodeId
          </label>
          <span
            class="w-1/2
              capitalize
              text-black dark:text-white"
          >
            4
          </span>
        </div>
        <div
          class="flex"
        >
          <label
            class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
          >
            director
          </label>
          <span
            class="w-1/2
              capitalize
              text-black dark:text-white"
          >
            George Lucas
          </span>
        </div>
        <div
          class="flex"
        >
          <label
            class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
          >
            producer
          </label>
          <span
            class="w-1/2
              capitalize
              text-black dark:text-white"
          >
            Gary Kurtz, Rick McCallum
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ConnectionDetails component > should render planet details 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="
        absolute top-2 left-2
        text-sm
        rounded-md
        border-2
        border-swblue dark:border-swyellow
        max-w-60 min-w-60
        bg-swgray/90 dark:bg-swblack/90
      "
      >
        <div
          class="p-2
        font-semibold
        text-white dark:text-swblack
        bg-swblue dark:bg-swyellow
        flex justify-between items-stretch
      "
        >
          <span>
            Tatooine
          </span>
          <i
            class="fa fa-globe self-center"
          />
        </div>
        <div
          class="p-2"
        >
          <div
            class="flex"
          >
            <label
              class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
            >
              type
            </label>
            <span
              class="w-1/2
              capitalize
              text-black dark:text-white"
            >
              Homeworld
            </span>
          </div>
          <div
            class="flex"
          >
            <label
              class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
            >
              terrain
            </label>
            <span
              class="w-1/2
              capitalize
              text-black dark:text-white"
            >
              desert
            </span>
          </div>
          <div
            class="flex"
          >
            <label
              class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
            >
              climate
            </label>
            <span
              class="w-1/2
              capitalize
              text-black dark:text-white"
            >
              arid
            </span>
          </div>
          <div
            class="flex"
          >
            <label
              class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
            >
              surfaceWater
            </label>
            <span
              class="w-1/2
              capitalize
              text-black dark:text-white"
            >
              1
            </span>
          </div>
          <div
            class="flex"
          >
            <label
              class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
            >
              gravity
            </label>
            <span
              class="w-1/2
              capitalize
              text-black dark:text-white"
            >
              1 standard
            </span>
          </div>
          <div
            class="flex"
          >
            <label
              class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
            >
              population
            </label>
            <span
              class="w-1/2
              capitalize
              text-black dark:text-white"
            >
              200000
            </span>
          </div>
          <div
            class="flex"
          >
            <label
              class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
            >
              diameter
            </label>
            <span
              class="w-1/2
              capitalize
              text-black dark:text-white"
            >
              10465
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="
        absolute top-2 left-2
        text-sm
        rounded-md
        border-2
        border-swblue dark:border-swyellow
        max-w-60 min-w-60
        bg-swgray/90 dark:bg-swblack/90
      "
    >
      <div
        class="p-2
        font-semibold
        text-white dark:text-swblack
        bg-swblue dark:bg-swyellow
        flex justify-between items-stretch
      "
      >
        <span>
          Tatooine
        </span>
        <i
          class="fa fa-globe self-center"
        />
      </div>
      <div
        class="p-2"
      >
        <div
          class="flex"
        >
          <label
            class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
          >
            type
          </label>
          <span
            class="w-1/2
              capitalize
              text-black dark:text-white"
          >
            Homeworld
          </span>
        </div>
        <div
          class="flex"
        >
          <label
            class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
          >
            terrain
          </label>
          <span
            class="w-1/2
              capitalize
              text-black dark:text-white"
          >
            desert
          </span>
        </div>
        <div
          class="flex"
        >
          <label
            class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
          >
            climate
          </label>
          <span
            class="w-1/2
              capitalize
              text-black dark:text-white"
          >
            arid
          </span>
        </div>
        <div
          class="flex"
        >
          <label
            class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
          >
            surfaceWater
          </label>
          <span
            class="w-1/2
              capitalize
              text-black dark:text-white"
          >
            1
          </span>
        </div>
        <div
          class="flex"
        >
          <label
            class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
          >
            gravity
          </label>
          <span
            class="w-1/2
              capitalize
              text-black dark:text-white"
          >
            1 standard
          </span>
        </div>
        <div
          class="flex"
        >
          <label
            class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
          >
            population
          </label>
          <span
            class="w-1/2
              capitalize
              text-black dark:text-white"
          >
            200000
          </span>
        </div>
        <div
          class="flex"
        >
          <label
            class="w-1/2
              capitalize
              font-semibold
              text-swblue dark:text-swyellow"
          >
            diameter
          </label>
          <span
            class="w-1/2
              capitalize
              text-black dark:text-white"
          >
            10465
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ConnectionDetails component > should render when clicked outside 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
